llama-cpp.nix

```diff
@@ -0,0 +1,118 @@
+{ lib
+, stdenv
+, cmake
+, ninja
+, pkg-config
+, symlinkJoin
+, python3
+, openmpi
+, openblas
+, cudaPackages
+, rocmPackages
+, clblast
+, Accelerate ? null
+, MetalKit ? null
+, CoreVideo ? null
+, CoreGraphics ? null
+, useOpenCL ? false
+, useCuda ? false
+, useRocm ? false
+}:
+
+let
+  inherit (lib) optional optionals versionOlder;
+  isDefault = !useOpenCL && !useCuda && !useRocm;
+
+  # Give a little description difference between the flavors.
+  descriptionSuffix = if useOpenCL then
+    " (OpenCL accelerated)"
+  else if useCuda then
+    " (CUDA accelerated)"
+  else if useRocm then
+    " (ROCm accelerated)"
+  else if (MetalKit != null) then
+    " (MetalKit accelerated)"
+  else "";
```

Comment:
Nit: `optionalString useXXX "..." + optionalString useYYY "..."` might look cleaner in the end

Reply:
I decided after trying it out to keep what I originally had. Another PR can change it.